implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceMeterRegistryImplBase.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2021 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.dynatrace;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.util.NamedThreadFactory;
+import io.micrometer.core.ipc.http.HttpSender;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Base class for implementations of the {@link io.micrometer.dynatrace.DynatraceMeterRegistry} delegate.
+ *
+ * @author Georg Pirklbauer
+ */
+public abstract class DynatraceMeterRegistryImplBase {
```

Comment:
To me it's not clear what is the purpose of this class.
Is this a `MeterRegistry` that will be used as a delegate or is it a client?

In the former case Could you please rename this class to something better, e.g.: `AbstractDynatraceMeterRegistry` and extend `StepMeterRegistry`?
Impl and Base seems weird to me, e.g.: if you want to implement cars your interface would be a `Car`, that you can specialize further by having an `AbstractCar implements Car` maybe a `SimpleCar extends AbstractCar` and a `SpecialCar extends AbstractCar`.

In the latter, could you please rename this class to something better, e.g.: `DynatraceV1Client` and clean it up a little so that it will look like a client not a registry?

Reply:
Renamed the abstract base class to `AbstractDynatraceExporter` and the implementation to `ApiV1DynatraceExporter`. I also renamed the function for exporting data to `export` so it's distinguishable from the `publish` function on the registry.

Reply:
Moved it again, now to `DynatraceExporterV1`, which seems more readable to me.